File: Data Structures & Algorithms/maximum-subarray/submission-1.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        for (int i = 0; i < nums.length; i++){
            int currentSum = nums[i];
            if(i == nums.length - 1){
                maxSum = Math.max(currentSum, maxSum);
                break;
            }
            for (int j = i+1; j < nums.length; j++){
                currentSum = currentSum + nums[j];
                maxSum = Math.max(maxSum, currentSum);
            }
        }
        return maxSum;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        minSpeed = 0
        left = 1
        right = max(piles) 
        while left <= right:
            k = (left + right) // 2
            total = 0
            for pile in piles:
                total += math.ceil(float(pile) / k)
            if total <= h:
                minSpeed = k
                right = k - 1 
            if total > h:
                left = k + 1
        return minSpeed 